### ISSUE TITLE
Cant discover Devices on Arch linux

### ISSUE THREAD

**Issue by simann** [serialport/node-serialport#2371]
### SerialPort Version

10.0.0

### Electron Version

15.3.0

### Platform

Linux 35v211210.mplocal 5.15.0-rc2-gentoo-dist #3 SMP Wed Sep 22 13:40:34 -00 2021 x86_64 GNU/Linux

### Architecture

x86_64

### What steps will reproduce the bug?

Minimal example to reproduce:

```
const { app } = require('electron');
const SerialPort = require("serialport");

app
  .whenReady()
  .then(ready);

function ready(){
  
  SerialPort.list()
    .then((ports) => {
      console.log(ports);
    });
}
```

Necessary files included:
[app.zip](https://github.com/serialport/node-serialport/files/7725871/app.zip)

### What happens?

`Serialport.list()` returns an empty array.

The connected device is an arduino.

Take a look at the following video

https://user-images.githubusercontent.com/28533890/146344577-c27e1728-39a3-46ee-af41-29157c5c39bc.mp4

### What should have happened?

 `Serialport.list()` to returns an array with 1 Element. The element should contain information about the Arduino

### Additional information

I tried with `nodejs`  alone aswell, and the same behaviour. No devices found. Node version is 17.2.0

On my windows machine everything is working as expected:
![electron serial](https://user-images.githubusercontent.com/28533890/146345482-ee33bc02-7d46-4d19-81ac-aba98deb9aee.png)

Besides that i had no problem installing the package on windows or linux.

Last working version was:
9.4 in combination with electron 12.0.07

Any hint, advice or confirmation would be appreciated ❤️

**Comment by GazHank** [comment]
Thanks for the info.  Could you confirm if the successful operation with the older serialport version, and electron version also executed with the same gentoo release and udev rules (just want to rule out other variances that could affect things).

Based on your distro I'm assuming you are using glibc, so that shouldn't be an issue.

**Comment by simann** [comment]
Im trying to rebuild serialport with electron 12.0.7, serialport 9.2.8. and electron-rebuild 3.2.5

Unfortunately building native modules is a real PITA. Currently i got errors about "openssl_fips" missing. Didnt figure out whats wrong yet but i keep trying.

I will keep you informed😄

**Comment by robertsLando** [comment]
Quite sure that if it's wortking with serialport < 9.1 it's related to this: https://github.com/serialport/node-serialport/issues/2266

**Comment by simann** [comment]
I had some time to make some more tests:

Im only talking about the plain nodejs version!

I tried https://github.com/serialport/node-serialport/issues/2266#issuecomment-895771364

no devices were found

Next thing was installing `npm install serialport@9.0.0 --build-from-source`

no devices were found

I dont think the Issue https://github.com/serialport/node-serialport/issues/2266 is the same, because in the mentioned issue serialport below 9.1.0 was working. On my distro i installed 9.0 from source and it still didnt work.

If you need any logs or information, I am happy to get it for you.

Side Note:
I am on  a 1 week vacation starting tomorrow.

**Comment by reconbot** [comment]
Lots of changes since 10.0.0 - also it seems we've moved from electron to node? I'm confused, but if this is still an issue - could you clarify a bit more?

**Comment by simann** [comment]
You are correct.

I switched to node because i wanted to erase the possibility of electron beeing the culprit.

If it works in node, then i would try it in electron again.

Im out of office today but i will try tomorrow with the current serialport version.

Thanks for the efforts :)

**Comment by simann** [comment]
As i said i will test again and the result is negativ. Still no devices are found.

As always i try to provide as much imformation as possible, in case anyone finds a mistake please let me know 😃 

I used the new serialport version 10.2.1:
![image](https://user-images.githubusercontent.com/28533890/152515106-492c0587-4199-4afd-9772-f852928c87e8.png)

On node v17:
![image](https://user-images.githubusercontent.com/28533890/152516242-9e9089d2-9d05-4626-86b8-3aca4689d9a8.png)


This is the file i am executing:
![image](https://user-images.githubusercontent.com/28533890/152515209-6d467a27-5d9a-4123-aeda-02fe9df4b9fc.png)

I had to make a minor change regarding the import of the package, to make it work with the new version.

```
old:
const SerialPort = require('serialport');

new:
const { SerialPort } = require('serialport');
```

This is the result:
![image](https://user-images.githubusercontent.com/28533890/152515379-b396e191-5d39-4dab-b1bf-f203f3e354ca.png)

To be save i tried rebuilding the package
![image](https://user-images.githubusercontent.com/28533890/152516600-1fd28227-3bec-4f01-bf45-1c0df312420c.png)

No success either 😞 

Can i confirm that there is a device attached?
Yes i can, these are the dmesg logs:
![image](https://user-images.githubusercontent.com/28533890/152515547-20af960f-d369-4f00-8258-e9fabdbc1724.png)

What Glibc Version i am using?
![image](https://user-images.githubusercontent.com/28533890/152516080-50c58a88-2ccb-4a6d-bee7-36be844bb1f9.png)

To be more specific, pacman installs the version 2.33-5 [Link](https://archlinux.org/packages/core/x86_64/glibc/) currently.

This is my distro:
![image](https://user-images.githubusercontent.com/28533890/152516442-78c73ee1-480c-4583-bc56-a80bf8900c06.png)

These are the result with an older version- 9.0.0:

![image](https://user-images.githubusercontent.com/28533890/152517202-b982fb83-d7fb-428c-a3a7-bdce39c4f8a2.png)

No success aswell

Kind of clueless tbh, looks like none of the serialport packages are working on the distro 😞 

Sincerely
Simon

**Comment by reconbot** [comment]
What's the path it's supposed to find?

**Comment by simann** [comment]
The path in the linux fs is:
![image](https://user-images.githubusercontent.com/28533890/152540945-fd7b3ab8-0d74-44dc-a157-0095037d7441.png)

/dev/input/event16

respectively:
![image](https://user-images.githubusercontent.com/28533890/152541249-056b1337-758a-4710-b212-42930e8c0a45.png)

/dev/input/by-id/usb-arduino_LLC...

### Further testing:

On my windows machine i find the same device.
![image](https://user-images.githubusercontent.com/28533890/152541393-a458c375-4998-48bb-a1f5-3051ce85ba7a.png)

Besides that i have an spare ubuntu machine flying around the office and i tried the provided code and it works:

Distro:
![image](https://user-images.githubusercontent.com/28533890/152548650-8ac69377-c63f-4951-bd60-07ed79cfc72b.png)

SerialPort 9.0.0:
![image](https://user-images.githubusercontent.com/28533890/152544478-cc9bf3e2-9f5d-4d06-a145-f63e79381ab8.png)

SerialPort 10.0.0:
![image](https://user-images.githubusercontent.com/28533890/152546717-8ac5e65f-d5d6-4c20-8f60-59ff2f2d1114.png)

SerialPort 10.2.1:
![image](https://user-images.githubusercontent.com/28533890/152548527-298b1660-06d3-4361-96f4-2a841ba712df.png)

GLIBC on Ubuntu:
![image](https://user-images.githubusercontent.com/28533890/152549575-a3426a20-bc62-4d69-8e65-4c38722cf42a.png)


These are my results:
Windows: ✔️ 
Linux - Ubunutu: ✔️ 
Linux: - Arch: ❌

**Comment by simann** [comment]
Are there any updates , e.g. should i test with the newest version?

Is anyone been able to confirm this?

**Comment by simann** [comment]
I have spent yesterday, trying to figure out what went wrong and i finally found the culprit.

tldr: nohing is wrong with serialport, it was all my fault

The error was due to changes i made when i was building the kernel. The kernel was based on an gentoo kernel but that didnt work out of the box with the hardware we had. I build different kernels with more or less features enabled, until it worked.

Unfortunaltely i removed/lost the USB_SERIAL_PL2303 driver in that process and because of this the serial device could not been found.

I rebuild the kernel today and now it works.

Thanks for all the effort  tho 👍
